datafusion/sqllogictest/test_files/insert_to_external.slt

```diff
@@ -57,7 +57,7 @@ CREATE EXTERNAL TABLE dictionary_encoded_parquet_partitioned(
   b varchar,
 ) 
 STORED AS parquet
-LOCATION 'test_files/scratch/insert_to_external/parquet_types_partitioned'
+LOCATION 'test_files/scratch/insert_to_external/parquet_types_partitioned/'
```

Comment:
This is necessary because now that the directories no longer exist it falls back to using the trailing / to determining if a directory or file is desired.

Reply:
Now that inserting (appending) to an individual file is no longer a thing `ListingTable` needs to concern itself with, I wonder if the concept of a "single file" table could be completely removed. I.e. whether there is a trailing slash or not, the `LOCATION` is interpreted as the directory where the data files will be written to / read from.

Reply:
Yes, this is definitely the eventual goal, is this something you would be interested in working on?

Reply:
The usecase of writing to `/foo/bar/1.parquet` (a single file) is important, but perhaps that is triggered by the name ending in `.parquet` 🤔 

I think that would still be achievable with the proposal above, I just wanted to point it out

Reply:
Yes, COPY has a SINGLE_FILE_OUTPUT option that gates this, but for `CREATE EXTERNAL TABLE` single files don't support insert

Reply:
Ticket for this https://github.com/apache/arrow-datafusion/issues/8548

Reply:
I'm happy to take a stab at removing the single file table option next week, when I also plan to look at adding support for writing to arrow files.